File: Lavcode.Service.Api/Dtos/UpdatePasswordDto.cs
using Newtonsoft.Json;
using System.Collections.Generic;

namespace Lavcode.Service.Api.Dtos
{
    public class UpdatePasswordDto
    {
        [JsonProperty("folderId")]
        public string FolderId { get; set; }

        [JsonProperty("title")]
        public string Title { get; set; }

        [JsonProperty("value")]
        public string Value { get; set; }

        [JsonProperty("remark")]
        public string Remark { get; set; }

        [JsonProperty("order")]
        public int Order { get; set; }

        [JsonProperty("icon")]
        public UpsertIconDto Icon { get; set; }

        [JsonProperty("keyValuePairs")]
        public IList<UpsertKeyValuePairDto> KeyValuePairs { get; set; }
    }
}

File: src/lavcode-node/src/actions/password/dtos/update-password.dto.ts
import { V } from "@halsp/validator";
import { UpsertIconDto } from "../../../dtos/upsert-icon.dto";
import { UpsertKeyValuePair } from "./upsert-key-value-pair.dto";

export class UpdatePasswordDto {
  @V().Required()
  folderId!: string;
  @V().IsOptional()
  title?: string;
  @V().IsOptional()
  value?: string;
  @V().IsOptional()
  remark?: string;
  @V().IsNumber().Required()
  order!: number;

  @V().IsOptional()
  icon!: UpsertIconDto;
  @V().IsOptional().Items(UpsertKeyValuePair)
  keyValuePairs!: UpsertKeyValuePair[];
}
